ush/python/pygw/src/tests/test_executable.py

```diff
@@ -0,0 +1,46 @@
+import os
+from pathlib import Path
+import pytest
+from pygw.executable import Executable, which, CommandNotFoundError
+
+
+script = """#!/bin/bash
+echo ${USER}
+"""
+
+
+def test_executable(tmp_path):
+    whoami = os.environ['USER']
+
+    test_file = tmp_path / 'whoami.x'
+    Path(test_file).touch(mode=0o755)
+    with open(test_file, 'w') as fh:
+        fh.write(script)
+
+    cmd = Executable(str(test_file))
+    assert cmd.exe == [str(test_file)]
+
+    stdout_file = tmp_path / 'stdout'
+    stderr_file = tmp_path / 'stderr'
+    cmd(output=str(stdout_file), error=str(stderr_file))
+    with open(str(stdout_file)) as fh:
+        assert fh.read() == whoami + '\n'
+
```

Comment:
Do we want additional tests of `Executable` where we set an environment variable, provide an input file, and stderr has non-null contents?

Reply:
More tests would definitely be great.  Feel free to conjure up some or we can add them in a future PR.